lib/_http_outgoing.js

```diff
@@ -388,7 +419,10 @@ OutgoingMessage.prototype.getHeader = function getHeader(name) {
 
   if (!this._headers) return;
 
-  return this._headers[name.toLowerCase()];
+  var entry = this._headers[name.toLowerCase()];
```

Comment:
I don't recall specifically if I tested it in here or not, but in some places such functions didn't seem to perform as well for whatever reason.